SimplifiedTests/NYPLAxisTests/NYPLAxisServiceTests.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  NYPLAxisServiceTests.swift
+//  OpenEbooksTests
+//
+//  Created by Raman Singh on 2021-05-14.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import XCTest
+@testable import SimplyE
+
+class NYPLAxisServiceTests: XCTestCase {
+  
+  lazy private var downloadsDirectory: URL = {
+    let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+    let documentsDirectory = paths[0]
+    return documentsDirectory.appendingPathComponent("NYPLAxisLicenseServiceTests")
+  }()
+  
+  lazy private var book: NYPLBook = {
+    let acquisitions = [NYPLFake.genericAcquisition.dictionaryRepresentation()]
+    
+    return NYPLBook(dictionary: ["acquisitions": acquisitions,
+                                 "categories" : ["Fantasy"],
+                                 "id": "666",
+                                 "title": "The Lord of the Rings",
+                                 "updated": "2020-09-08T09:22:45Z"
+    ])!
+  }()
+  
+  private let itemDownloader = NYPLAxisItemDownloader()
+  private let downloadBroadcaster = NYPLAxisBookDownloadBroadcasterMock()
+  
+  func testAxisServiceShouldFailUponLicenseDownloadFailure() {
+    let downloadFailedExpectation = XCTestExpectation(
+      description: "Failed license process should result in failed download!")
+    let downloadSucceededExpectation = XCTestExpectation(
+      description: "Download shouldn't succeed upon license failure")
+    
+    downloadSucceededExpectation.isInverted = true
+    
+    let axisService = createAxisService(licenseService: failingLicenseService,
+                                        metadataService: succeedingMetadataService,
+                                        packageService: succeedingPackageService)
+    
+    let task = URLSessionDownloadTask()
+    
+    downloadBroadcaster.downloadFailed = {
+      downloadFailedExpectation.fulfill()
+    }
+    
+    downloadBroadcaster.downloadSuccessful = {
+      downloadSucceededExpectation.fulfill()
```

Comment:
wouldn't this mean that we have to wait 10 seconds for this inverted expectation to be... not fulfilled? I haven't used inverted expectations often. 

wouldn't it be simpler to just do a XCFail() inside the success block instead? Same for the other tests, wherever we have explicit fail / success blocks.

Reply:
Agreed. Fixed.